### ISSUE TITLE
bug: Build fails on brand-new machine due to missing type definitions for date-fns package

### ISSUE THREAD

**Issue by rmunn** [sillsdev/web-languageforge#1404]
**Describe the bug**
On a brand-new Windows 10 machine with WSL 2 and Docker installed, `make` fails to build. The error message is from Typescript complaining that `node_modules/date-fns/index.js` implicitly has an `any` type. The recommended fix was to install the `@types/date-fns` module. When I did so, that package informed be that date-fns has its own types, so `@types/date-fns` is a stub and shouldn't need to be installed.

After a bit of experimentation, I discovered that the version of NPM that had been installed with Node 16 on this new machine was NPM 8. I downgraded to NPM 7 by doing `npm install -g npm@7` and then the date-fns types could be found. So there's something about NPM 8 that breaks with date-fns 2.23.

We'll either need to upgrade date-fns to the latest release (2.28), or else add instructions to the DEVELOPERS.md docs to say "Make sure that NPM 7, not NPM 8, is installed."

**Comment by megahirt** [comment]
Assuming that date-fns can be upgraded without much effort, we should stick
with NPM 8.

**Comment by rmunn** [comment]
Could not reproduce on a different machine; this appears to have been an artifact of the weird setup of one particular machine, where running `node` from inside WSL was actually picking up a Node version installed on the Windows side (in `C:\Program Files\whatever`). When I tried to reproduce this issue on a different machine running WSL where Node was *not* installed on the Windows side, the current code ran just fine as-is.

Closing as not reproducible. We'll still want to upgrade date-fns to the latest version as part of keeping all of our NPM packages up-to-date, but it doesn't need special treatment.
